addons/l10n_vn_edi/models/account_move.py

```diff
@@ -0,0 +1,719 @@
+        string='Secret Code',
+        help='Secret code that can be used by a customer to lookup an invoice on SInvoice.',
+        copy=False,
+        readonly=True,
+    )
+    l10n_vn_edi_issue_date = fields.Datetime(
+        string='Issue Date',
+        help='Date of issue of the invoice on the e-invoicing system.',
+        copy=False,
+        readonly=True,
+    )
+    l10n_vn_edi_sinvoice_file_id = fields.Many2one(
+        comodel_name='ir.attachment',
+        compute=lambda self: self._compute_linked_attachment_id('l10n_vn_edi_sinvoice_file_id', 'l10n_vn_edi_sinvoice_file'),
+        depends=['l10n_vn_edi_sinvoice_file'],
+        export_string_translation=False,
+    )
+    l10n_vn_edi_sinvoice_file = fields.Binary(
+        string='SInvoice json File',
+        copy=False,
+        export_string_translation=False,
+    )
+    l10n_vn_edi_sinvoice_xml_file_id = fields.Many2one(
+        comodel_name='ir.attachment',
+        compute=lambda self: self._compute_linked_attachment_id('l10n_vn_edi_sinvoice_xml_file_id', 'l10n_vn_edi_sinvoice_xml_file'),
+        depends=['l10n_vn_edi_sinvoice_xml_file'],
+        export_string_translation=False,
+    )
+    l10n_vn_edi_sinvoice_xml_file = fields.Binary(
+        string='SInvoice xml File',
+        copy=False,
+        export_string_translation=False,
+    )
+    l10n_vn_edi_sinvoice_pdf_file_id = fields.Many2one(
+        comodel_name='ir.attachment',
+        compute=lambda self: self._compute_linked_attachment_id('l10n_vn_edi_sinvoice_pdf_file_id', 'l10n_vn_edi_sinvoice_pdf_file'),
+        depends=['l10n_vn_edi_sinvoice_pdf_file'],
+        export_string_translation=False,
+    )
+    l10n_vn_edi_sinvoice_pdf_file = fields.Binary(
+        string='SInvoice pdf File',
+        copy=False,
+        export_string_translation=False,
+    )
+    # Replacement/Adjustment fields
+    l10n_vn_edi_agreement_document_name = fields.Char(
+        string='Agreement Name',
+        copy=False,
+    )
+    l10n_vn_edi_agreement_document_date = fields.Datetime(
+        string='Agreement Date',
+        copy=False,
+    )
+    l10n_vn_edi_adjustment_type = fields.Selection(
+        string='Adjustment type',
+        selection=[
+            ('1', 'Money adjustment'),
+            ('2', 'Information adjustment'),
+        ],
+        copy=False,
+    )
+    # Only used in case of replacement invoice.
+    l10n_vn_edi_replacement_origin = fields.Many2one(
+        comodel_name='account.move',
+        string='Replacement of',
+        copy=False,
+        readonly=True,
+        check_company=True,
+        export_string_translation=False,
+    )
+
+    @api.depends('l10n_vn_edi_state')
+    def _compute_show_reset_to_draft_button(self):
+        # EXTEND 'account'
+        super()._compute_show_reset_to_draft_button()
+        self.filtered(lambda m: m.l10n_vn_edi_invoice_number and m.l10n_vn_edi_state != 'cancelled').show_reset_to_draft_button = False
```

Comment:
But it's not the same thing?
At the moment, reset to draft is shown:
- While it has not been sent
- If it has been sent, but then canceled

You lose that second option with your solution. I can accept it, but then please tell me why I shouldn't allow that.

> m.l10n_vn_edi_state != 'cancelled' is a bad idea.
The point is that I intended to allow reset to draft if the invoice has been fully canceled. I don't see how adding a new state would break that unless that state replaces canceled/...